File: Triplet Sum in Array - GFG/triplet-sum-in-array.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution{
    public:
    //Function to find if there exists a triplet in the 
    //array A[] which sums up to X.
    bool find3Numbers(int a[], int n, int X)
    {
        //Your Code Here
       
    
        sort(a,a+n);
        for(int i = 0 ; i < n ; i++){
        int j = i+1;
        int k = n-1;
        while(j<k){
            if(a[i]+a[j]+a[k]==X){
            return true;
            }
            else if(a[i]+a[j]+a[k]<X){
                j++;
            }
            else{
                k--;
            }
        }
        }
        return false;
    }

};

//{ Driver Code Starts.

int main()
{
	int T;
	cin>>T;
	while(T--)
	{
		int n,X;
		cin>>n>>X;
		int i,A[n];
		for(i=0;i<n;i++)
			cin>>A[i];
		Solution ob;
        cout <<  ob.find3Numbers(A, n, X) << endl;
    }
}

// } Driver Code Ends

File: Triplet Sum in Array - GFG/triplet-sum-in-array.py
#User function Template for python3
class Solution:
     
    #Function to find if there exists a triplet in the 
    #array A[] which sums up to X.
    def find3Numbers(self,a, n, X):
        # Your Code Here
        a = sorted(a)
        for i in range(n):
            j = i+1
            k = n-1
            while(j<k):
                if a[i]+a[j]+a[k]==X:
                    return True
                elif a[i]+a[j]+a[k]<X:
                    j += 1
                else:
                    k -=1
        return False
                    


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        n,X=map(int,input().strip().split())
        A=list(map(int,input().strip().split()))
        ob=Solution()
        if(ob.find3Numbers(A,n,X)):
            print(1)
        else:
            print(0)
# } Driver Code Ends